### PR TITLE
fix: translation for pages

### PR DESCRIPTION
**Pull request by amanuela97**
## translated paths
- src/use/profiles
- src/pages/events/*
- src/pages/posts/*
- src/pages/community.vue
- src/use/accounts
- src/use/profiles

## How to test
- check out the corresponding URL using the demo app

Demo Link: https://wedance-amanuela97.netlify.app/

### CODE REVIEW

File: locales/en.yml [hunk 1]
```diff
@@ -104,7 +159,7 @@ share.title: Share
 share.copy: Copy Link
 footer.about: About us
 footer.support: Support
-footer.privacy: Privacy
+footer.privacy: PrivacyS
```
Comment (best-lucky1030) [inline locales/en.yml:162]: why did you add `S` at the end of word?
`PrivacyS` ?

**Review by best-lucky1030 — COMMENTED** [review]
it's typo or not?

**Comment by razbakov** [comment]
Please also include link to netlify demo page, where we can test it

File: locales/en.yml (hunk 1 above)
Comment (amanuela97) [reply locales/en.yml:162]: yes, it is a typo

File: pages/events/_id/dashboard.vue [hunk 2]
```diff
@@ -37,20 +39,20 @@
...
-          label="Send emails"
+          :label="$t('events.emails.label')"
           class="border-b"
           @click="composeEmail"
         />
         <TButton
           type="context"
           :to="`/events/${item.id}/`"
-          label="View Event"
+          :_label="$t('events.view.label')"
```
[the remark this answers is not in the document]
Comment (amanuela97) [reply pages/events/_id/dashboard.vue:55]: another typo

File: pages/events/_id/dashboard.vue [hunk 3]
```diff
@@ -1,24 +1,26 @@
...
   >
-    Only event owner can access this area.
+    {{ $t('events.dashboard.description') }}
   </main>
   <div v-else>
     <TPopup v-if="compose" title="Compose email" @close="compose = false">
       <div class="max-w-lg">
         <div class="text-xs mt-4">
-          {{ selectedParticipantsList.length }} recipients:
+          {{ selectedParticipantsList.length }} {{ $t('recipients') }}:
```
[the remark this answers is not in the document]
Comment (amanuela97) [reply pages/events/_id/dashboard.vue:16]: alright, will look into it

File: pages/community.vue [hunk 4]
```diff
@@ -1,14 +1,14 @@
...
+    <THeader :title="$t('community.title')">
       <TButton type="nav" icon="search" to="/search" />
     </THeader>
 
     <div>
       <div v-if="uid && response.facets" class="mb-4 gap-2 flex flex-wrap p-4">
         <t-rich-select
           v-model="radius"
-          placeholder="Radius"
+          :placeholder="$t('community.radius.select')"
```
Comment (best-lucky1030) [inline pages/community.vue:11]: all of texts are from locales?

**Review by best-lucky1030 — COMMENTED** [review]
looks all good.

File: pages/community.vue (hunk 4 above)
Comment (amanuela97) [reply pages/community.vue:11]: yes

File: components/TField.vue [hunk 5]
```diff
@@ -204,6 +203,17 @@ export default {
 
       return map[this.type] || TInput
     },
+    changeAttrs(attrs) {
```
Comment (razbakov) [inline components/TField.vue:206]: i wouldn't use magic, just use attrs always as keys for translations

File: pages/events/_id/dashboard.vue [hunk 6]
```diff
@@ -37,20 +46,20 @@
       <TDropdown>
         <TButton
           type="context"
-          label="Add participant"
+          :label="$t('events.dashboard.participant-label')"
```
Comment (razbakov) [inline pages/events/_id/dashboard.vue:49]: There is a mess in naming. Let's pick one style:
1) participant-label
2) participantlabel
3) participant.label
4) participantLabel

I prefer 3d.

Comment (razbakov) [reply pages/events/_id/dashboard.vue:49]: or in this case it could be `events.dashboard.menu.addParticipant`

File: pages/events/_id/dashboard.vue [hunk 7]
```diff
@@ -195,7 +215,7 @@
                     <TMenu2
                       v-if="!item.partnerId"
                       wrapped
-                      label="Link"
+                      :label="$t('events.dashboard.link-label')"
```
Comment (razbakov) [inline pages/events/_id/dashboard.vue:218]: Maybe `events.dashboard.actions.link`?

File: pages/events/_id/index.vue [hunk 8]
```diff
@@ -174,40 +179,39 @@
...
+              :submit-label="$t('events.register.submitlabel')"
               class="mt-4 space-y-4"
               @save="reserve"
             >
               <template v-if="!uid" slot="buttons">
                 <TButton
                   allow-guests
                   :to="`/signin?target=${this.$route.fullPath}`"
-                  label="Login"
+                  :label="$t('events.login.btn')"
```
Comment (razbakov) [inline pages/events/_id/index.vue:190]: I would re-use existing login translation

File: pages/posts/_id/share.vue [hunk 9]
```diff
@@ -6,7 +6,7 @@
     :description="getExcerpt(post.description)"
     :photo="post.cover"
     :styles="post.styles"
-    price="Read more on wedance.vip"
+    :price="$t('posts.share.price')"
```
Comment (razbakov) [inline pages/posts/_id/share.vue:9]:
```suggestion
    :price="$t('posts.share.readMore')"
```

**Review by razbakov — CHANGES_REQUESTED** [review]

**Comment by razbakov** [comment]
@amanuela97  please read https://github.com/we-dance/platform/discussions/70

File: pages/events/_id/dashboard.vue (hunk 6 above)
Comment (Stefanacef) [reply pages/events/_id/dashboard.vue:49]: I also only use option 3

File: components/TField.vue (hunk 5 above)
Comment (razbakov) [inline components/TField.vue:206]: - use computed field instead of method
- name it `translatedAttrs`

File: components/TField.vue [hunk 10]
```diff
@@ -204,6 +203,17 @@ export default {
...
+      if (attrs && attrs.options) {
+        attrs.options.map(
+          (option) =>
+            this.$te(option.label) && (option.label = this.$t(option.label))
+        )
+      } else if (attrs && attrs.placeholder && this.$te(attrs.placeholder)) {
```
Comment (razbakov) [inline components/TField.vue:212]: `else if` means placaholder will be translated only if options are not set

File: components/TField.vue [hunk 11]
```diff
@@ -204,6 +203,17 @@ export default {
 
       return map[this.type] || TInput
     },
+    changeAttrs(attrs) {
+      if (attrs && attrs.options) {
+        attrs.options.map(
+          (option) =>
+            this.$te(option.label) && (option.label = this.$t(option.label))
```
Comment (razbakov) [inline components/TField.vue:210]: i wouldn't check if translation exists and just use translation

**Review by razbakov — CHANGES_REQUESTED** [review]

File: locales/en.yml [hunk 12]
```diff
@@ -161,6 +155,11 @@ profile.before.hometown: Where were you born and raised?
 profile.before.name: If you want to remain anonym use your first name.
 profile.before.bio: Short introduction on top of your photo
 profile.before.story: Tell your dance story. You can also use [widgets](https://wedance.vip/markdown), including images and videos
+profile.visibilityOptions.public: Public
+profile.visibilityOptions.Members: Members
```
Comment (razbakov) [inline locales/en.yml:159]: members is uppercase, public is lowercase. pick one style please.

File: locales/en.yml [hunk 13]
```diff
@@ -28,35 +30,32 @@ events.notgoing.btn: Not going
...
-events.dashboard.view-label: View Event
-events.dashboard.cancel-btn: Cancel
-events.dashboard.check-in: Check In
+events.dashboard.submit.label: Send
+events.dashboard.compose.label: Compose email
+events.dashboard.addGuest: Add Guest
+events.dashboard.participant.label: Add participant
+events.dashboard.email.label: Send emails
+events.dashboard.view.label: View Event
+events.dashboard.cancel.btn: Cancel
```
Comment (razbakov) [inline locales/en.yml:50]: `view.label` and `cancel.btn` are both buttons. pick one style please. i would just drop `btn` and `label`

File: locales/en.yml [hunk 14]
```diff
@@ -119,7 +113,7 @@ profile.objectivesList.teach: Teach together
 profile.objectivesList.project: Dance project
 profile.profileSorts.username: Username
 profile.profileSorts.registered: Registered
-profile.profileSorts.last-seen: Last seen
+profile.profileSorts.lastSeen: Last seen
 profile.placeholder.username: username
 profile.placeholder.whatsapp: whatsapp chat
 profile.placeholder.business: for business requests
```
Comment (razbakov) [inline locales/en.yml:119]: fieldname is missing in this key

File: locales/en.yml [hunk 15]
```diff
@@ -119,7 +113,7 @@ profile.objectivesList.teach: Teach together
 profile.objectivesList.project: Dance project
 profile.profileSorts.username: Username
 profile.profileSorts.registered: Registered
-profile.profileSorts.last-seen: Last seen
+profile.profileSorts.lastSeen: Last seen
 profile.placeholder.username: username
 profile.placeholder.whatsapp: whatsapp chat
```
Comment (razbakov) [inline locales/en.yml:118]: it's better when fieldname goes before attribute, i.e. `profile.whatsapp.placeholder`

**Review by razbakov — CHANGES_REQUESTED** [review]

File: components/TField.vue [hunk 16]
```diff
@@ -138,6 +137,23 @@ export default {
     showTips: false,
   }),
   computed: {
+    translatedAttrs() {
+      let translated = this.$attrs
+      if (translated && translated.options) {
```
Comment (razbakov) [inline components/TField.vue:142]:
```suggestion
      if (translated?.options) {
```

**Review by razbakov — CHANGES_REQUESTED** [review]

**Review by best-lucky1030 — COMMENTED** [review]
Good  👍

File: components/TField.vue [hunk 17]
```diff
@@ -138,6 +137,23 @@ export default {
...
+      let translated = this.$attrs
+      if (translated?.options) {
+        const options = translated.options.map((option) => {
+          const label = option.label
+          return this.$t(label)
+        })
+        translated = { ...translated, options }
+      }
+
+      if (translated && translated.placeholder) {
```
Comment (razbakov) [inline components/TField.vue:150]: same here

**Review by razbakov — CHANGES_REQUESTED** [review]

File: locales/en.yml [hunk 18]
```diff
@@ -173,16 +171,28 @@ sort.newest: Newest
 style.label: Style
 profile.username: Username
 account.email: Email
-account.password: Password
+account.password: Passwordd
```
Comment (razbakov) [inline locales/en.yml:174]: typo

File: use/accounts.js [hunk 19]
```diff
@@ -41,48 +41,48 @@ export const accountFields = [
     options: [
       {
         value: 'party',
-        label: 'Social/party events',
+        label: 'account.update.party',
```
Comment (razbakov) [inline use/accounts.js:44]: newsletter.party

File: use/accounts.js [hunk 20]
```diff
@@ -41,48 +41,48 @@ export const accountFields = [
...
     type: 'text',
     event: true,
   },
   {
     name: 'withPartner',
     type: 'select',
-    label: 'Do you have a partner?',
-    options: ['Yes', 'No'],
+    key: 'account.withPartner',
+    options: ['account.yes', 'account.no'],
```
Comment (razbakov) [inline use/accounts.js:74]: Those could be just `yes`, `no` and can be reused elsewhere

**Review by razbakov — CHANGES_REQUESTED** [review]
